Iss416

ridhi96 opened the pull request:
**Mobility metric pull request template**

Please include the following points in your PR:

1) A link to the issue that this PR relates to. You can bring up a list of suggested issues and pull requests within the repository by typing #. #416 

2) A description of the content in this pull request.  
- What was changed? 
  - Created quarto files for county and place levels containing the updates to social capital 2 metric (economic connectedness)
- What should the reviewer be focusing on?
  - Change in methodology for calculating metric for both county and place levels
  - Overall structuring of the metric from ZCTA to place level.
- Is there a logical order to review the files in?
  - You can get a better sense of the updates by reviewing the county file first.
3) Detail on any issues or flags that the metric reviewer/data-team should be aware of.
   - Metric was updated based on [this](https://urbanorg.app.box.com/notes/1623512687034) guidance.

jwalsh28 commented on 06_neighborhoods/social-capital/economic_connectedness_county_2022.qmd:
```diff
@@ -0,0 +1,386 @@
+        ),
+      # calculate lower bound from standard error
+      share_economic_connectedness_lb = 
+       share_economic_connectedness - ( qnorm(0.975) * ec_se_county_updated ),
```
Why are we multiplying the standard error by `qnorm(0.975)`? If there is a justification please add a note on this in the text above the code chunk.

ridhi96 replied:
I had brought this up during the data team meeting, and according to Aaron, Tina did this to calculate the confidence intervals which checks out with the functionality of `qnorm` as it finds the critical value. Here, we are finding the critical value for a 95% confidence interval. I had reservations about the underlying data being normal, but we are going with the CLT assumption. I can check with the team again if this makes sense.

awunderground replied:
`qnorm(0.975)` is approximately 1.96, which is the critical value we want to construct a 95% confidence interval. 

The data don't need to be normally distributed for the sampling distribution of the statistic to be approximately normal as long as the sample size is sufficiently large. This is one of the major insights of the central limit theorem. That said, I don't know the distribution of the underlying data or the nature of the statistic to guess how large is sufficiently large.

jwalsh28 commented on 06_neighborhoods/social-capital/economic_connectedness_city_2022.qmd:
````diff
@@ -0,0 +1,497 @@
+names(ec_zip_raw)
+
+```
+
````
Before heading to the crosswalk I think we should show the distriubtion of raw data by ZIP, important to have a gut check for any outliers.

jwalsh28 commented on 06_neighborhoods/social-capital/economic_connectedness_city_2022.qmd:
```diff
@@ -0,0 +1,497 @@
+                                ZCTAinPlace >= 0.5 ~ 1,
+                                ZCTAinPlace < 0.5 ~ 0)
+          )
+      
```
Two visuals I would like to see here:
1) Can we get the distriubtion of the ZCTAinPlace variable?
2) A bar chart showing the counts of ZCTAs that have 1, 0.5-1 and less than 0.5

jwalsh28 commented on 06_neighborhoods/social-capital/economic_connectedness_city_2022.qmd:
````diff
@@ -0,0 +1,497 @@
+        drop_na(ec_zip)
+## Observation: Number of obs dropped to 54042 after removing `NA` values for economic connectedness
+
+```
````
Can we show by state the count of dropped ZCTAs? Curious if there are any anomalies.

jwalsh28 commented on 06_neighborhoods/social-capital/economic_connectedness_city_2022.qmd:
````diff
@@ -0,0 +1,497 @@
+
+```
+
+Now, calculate a weighted average of economic connectedness for each Plac, weighted by the percentage area of the ZCTA in that place (i.e. the `ZCTAinPlace` variable).  
````
Typo: "Plac"

jwalsh28 commented on 06_neighborhoods/social-capital/economic_connectedness_city_2022.qmd:
```diff
@@ -0,0 +1,497 @@
+# and also include total ZCTAs in Place & how many of those partially fall outside the Place 
+
+
+wtd_merged_ec_city <- merged_ec_city |>
```
There is a lot happening in this chunk that I think deserves individual attention. Creation of the zip_total and zips_in varaibles is fairly complex, I would do this in a seperate step and add a bit more explenation.

ridhi96 replied:
I have added explanations for each of these code transformations. I think that leaving this chunk of code unchanged is fine because of the grouping level, otherwise will have to do more manipulations to get the data to the place level. I think `zip_total` could have been achieved without `num_ZCTAs_in_place` but it's alright, hopefully my explanation makes more sense. 

I also changed the way `num_ZCTAs_in_place` was computed to give more intuition as to what is happening here. Let me know if it makes sense now!

jwalsh28 commented on 06_neighborhoods/social-capital/economic_connectedness_city_2022.qmd:
````diff
@@ -0,0 +1,497 @@
+```{r}
+#| label: data-quality-flag
+
+# Data Quality 1 = 50% or more of the ZIPs fall mostly (>50%) in the census place 
````
Double-check this step for me. I don't believe the code is actually doing what this explenation says. The variable `zips_in` is the sum of `portion_in` which I believe is 1 if the entire ZCTA falls in the place and zero if not. To match the explenation given here for quality I believe you would have to swap the variable `mostly_in` for `portion_in` above in the code-chunk labeled collapse-estimates. Please give this a close look.

ridhi96 replied:
Thank you for pointing this one out! If you refer Tina's code, she left a comment that she never used `mostly_in` variable after creating it (I have left this comment in the `qmd` file too). Based on referring the original code, I think Tina intended to use `portion_in` here. I believe the confusion stems from the word `mostly` in the quality variable explanation which I have changed to `fully` now. 

The gist is that quality variable considers what percentage of ZCTAs are fully within the boundaries of a Place. So higher the ZCTAs that are encompassed by a Place, the more confidence we have in the transformed economic connectedness value linked with that place. Does this make sense?

jwalsh28 reviewed (changes requested):
Thanks for transitinoing this to QMD Rhidi, it is a lot easier to read through and evaluate now. I have mostly harmless comments throughout but a flag for the quality variable for teh city data - I believe there was a slight error here in the prior code.

ridhi96 commented on 06_neighborhoods/social-capital/economic_connectedness_county_2022.qmd:
```diff
@@ -0,0 +1,386 @@
+
+# Visualize economic connectedness before updating
+ 
+merged_ec |>
```
I have added an explanation of economic connectedness and interpretation of the metric values based on what's present in the UMF Dashboard as of now. This value is quite tricky to interpret because of the way Opportunity Insights describes it in their paper. I have added the simplest explanation from our dashboard and the definition. Please let me know if it's still not clear enough.

I have also added text to explain what the metric depicts after it is updated (basically why the values are now between 0 and 1).

ridhi96 commented on 06_neighborhoods/social-capital/economic_connectedness_county_2022.qmd:
```diff
@@ -0,0 +1,386 @@
+
+EXAMPLE: Fairfax County, VA
+
+-   Urban’s Mobility Metrics Economic Connectedness: 1.2
```
Yep added a comment in the code file.

ridhi96 commented:
Hi @jwalsh28! I updated the files based on your review and added responses to your comments. I have tried to provide answers to some of the questions you had about methodology. Please let me know if you have any more thoughts/questions. Thank you!

jwalsh28 reviewed (approved):
This looks good Ridhi! I left one comment open on the `qnorm` decision to see if Aaron has a chance to chime in. Otherwise I think this is complete.

jwalsh28 commented on 06_neighborhoods/social-capital/economic_connectedness_county_2022.html:
```diff
@@ -3476,7 +3476,7 @@ <h2 class="anchored" data-anchor-id="update-the-economic-connectedness-variable"
 <p>Next step is to create a data quality flag and calculate confidence intervals for the updated economic connectedness metric.</p>
 <p>The counties for which the share of economic connectedness value exists will have a data quality of <code>1</code> following the instructions on the UMF project <a href="https://github.com/UI-Research/mobility-from-poverty/wiki/3_Data-Quality-and-Confidence-Intervals%E2%80%AF">Wiki</a>. A data quality of <code>1</code> indicates that our confidence in this metric is strong based on the rigorou ...
-<p>We will calculate the 95% confidence interval for the metric <code>share_economic_connectedness</code> using the <code>qnorm</code> function and standard error variable <code>ec_se_county_updated</code>.</p>
+<p>We will calculate the 95% confidence interval for the metric <code>share_economic_connectedness</code> using the <code>qnorm</code> function and standard error variable <code>ec_se_county_updated</code>. The value of <code>qnorm(0.975)</code> is approximately 1.96, which is the critical value we want to construct a 95% confidence interval. Referencing the central limit theorem, we consider the ...
```
Looks great!

jwalsh28 reviewed (approved):
Thanks for adding this Ridhi!
